applications/dashboard/src/scripts/app/user-content/embeds/giphy.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license https://opensource.org/licenses/GPL-2.0 GPL-2.0
+ */
+
+import { registerEmbed, IEmbedData, FOCUS_CLASS } from "@dashboard/embeds";
+
+registerEmbed("giphy", giphyRenderer);
+
+/**
+ * Renders giphy embeds.
+ */
+export async function giphyRenderer(element: HTMLElement, data: IEmbedData) {
+    if (data.attributes.postID == null) {
+        throw new Error("Giphy embed fail, the post could not be found");
+    }
+
+    element.classList.add("embedGiphy");
+    element.style.width = `${data.width}px` || "100%";
```

Comment:
This won't quite work, will it? If `data.width` is empty, won't "px" still cause "100%" not to kick in?